tox.ini

```diff
@@ -1,48 +1,167 @@
 [tox]
 skipsdist=True
-envlist=lint-py36,py36-dj{111}-wag{23}
+# Temporarily disabling this to deliberately support conflicting versions
+# of django-treebeard and djangorestframework.
+# tox_pip_extensions_ext_venv_update=True
+# Run these envs when tox is invoked without -e
+envlist=lint-{current}, unittest-{current}
+
 
 [testenv]
+# Factored env for running most things with a matrix of Python, Django, and
+# Wagtail versions. Configuration of individual top-level factors like lint
+# and unittest is done in their respective configuration sections below.
+#
+# Factors:
+#   lint:     Lint Python files with flake8 and isort
+#   unittest: Run Python unittests
+#   current:  Use the current production Python, Django, Wagtail
+#   future:   Use the latest production Python, Django, Wagtail
+#
+# These factors are expected to combine into the follow generative environments:
+#
+#   lint-current
+#   lint-future
+#   unittest-current
+#   unittest-future
+#
+# These factors are expected to combine to be invoked with:
+#
+#   tox -e lint-current
+#   tox -e lint-future
+#   tox -e unittest-current
+#   tox -e unittest-future
+
+recreate=False
+
+whitelist_externals=echo
+
+basepython=
+    current:  {[current-config]basepython}
+    future:   {[future-config]basepython}
+
+deps=
+    current:  {[current-config]deps}
+    future:   {[future-config]deps}
+    lint:     {[lint-config]deps}
+    unittest: {[unittest-config]deps}
+
+passenv=
+    unittest:   {[unittest-config]passenv}
+
+setenv=
+    unittest: {[unittest-config]setenv}
+
+commands=
+    lint:     {[lint-config]commands}
+    unittest: {[unittest-config]commands}
+
+
+[lint-config]
+# Configuration necessary to lint Python files.
+# Note: This is not an env and will not run if invoked. Use:
+#
+#   tox -e lint-{current, future}
+#
+# To run Python linting.
+deps=
+    black
+    flake8
+    isort
+commands=
+    black --check teachers_digital_platform setup.py
+    flake8 teachers_digital_platform
+    isort --check-only --diff --recursive teachers_digital_platform
+
+
+[unittest-config]
+# Configuration necessary to run unittests.
+# Note: This is not an env and will not run if invoked. Use:
+#
+#   tox -e unittest-{current, future}
+#
+# To run unit tests.
 install_command=pip install -e ".[testing]" -U {opts} {packages}
+passenv=
+    TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH TEST_DATABASE_URL TEST_RUNNER
+setenv=
+    DJANGO_SETTINGS_MODULE=teachers_digital_platform.tests.settings
+    PYTHONPATH={toxinidir}/cfgov-refresh/cfgov:{env:PYTHONPATH:}
+    TEST_DATABASE_URL=postgres://cfpb@localhost/cfgov
```

Comment:
Let's leave `TEST_DATABASE_URL` out of this PR for now. We can resolve the testing issue related to it once cfgov-refresh has a solution from https://github.com/cfpb/cfgov-refresh/pull/5627/.
```suggestion
```

Reply:
Removed `TEST_DATABASE_URL`